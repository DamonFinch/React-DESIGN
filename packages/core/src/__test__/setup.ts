import '@testing-library/jest-dom'
import 'jest-styled-components'
import { cleanup } from '@testing-library/react'
import { afterEach } from 'vitest'

afterEach(cleanup)
