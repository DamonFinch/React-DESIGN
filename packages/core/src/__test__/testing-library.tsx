import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

function customRender(ui: React.ReactElement, options = {}) {
  return render(ui, {
    wrapper: ({ children }) => <ThemeProvider>{children}</ThemeProvider>,
    ...options,
  })
}

export * from '@testing-library/react'
export { customRender as render }

export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
