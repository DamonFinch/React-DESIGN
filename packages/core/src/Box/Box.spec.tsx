import { describe, it, expect } from 'vitest'
import { render } from '../__test__/testing-library'

import { Box } from '.'

describe('Box', () => {
  it('renders', () => {
    const { asFragment } = render(<Box />)
    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        <div
          class="cMczSr"
        >
          <div
            class="cETxcT"
          />
        </div>
      </DocumentFragment>
    `)
  })

  it('width and height props set width/height', () => {
    const { asFragment } = render(<Box width={1 / 2} height='50%' />)
    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        <div
          class="cMczSr"
        >
          <div
            class="gTAEUK"
            height="50%"
            width="0.5"
          />
        </div>
      </DocumentFragment>
    `)
    // expect(json).toHaveStyleRule('height', '50%')
    // expect(json).toHaveStyleRule('width', '50%')
  })

  it('m prop sets margin', () => {
    const { asFragment } = render(<Box m={2} />)
    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        <div
          class="cMczSr"
        >
          <div
            class="bAeKOk"
          />
        </div>
      </DocumentFragment>
    `)
    // expect(json).toHaveStyleRule('margin', theme.space[2])
  })

  it('maxHeight, maxWidth, minHeight, minWidth', () => {
    const { asFragment } = render(<Box maxHeight={250} maxWidth={250} minHeight={55} minWidth={55} />)
    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        <div
          class="cMczSr"
        >
          <div
            class="gXVrDr"
          />
        </div>
      </DocumentFragment>
    `)
    // expect(json).toHaveStyleRule('max-height', '250px')
    // expect(json).toHaveStyleRule('max-width', '250px')
    // expect(json).toHaveStyleRule('min-height', '55px')
    // expect(json).toHaveStyleRule('min-width', '55px')
  })

  it('p prop sets padding', () => {
    const { asFragment } = render(<Box p={2} />)
    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        <div
          class="cMczSr"
        >
          <div
            class="hgdKtj"
          />
        </div>
      </DocumentFragment>
    `)
    // expect(json).toHaveStyleRule('padding', theme.space[2])
  })

  it('color prop sets color', () => {
    const { asFragment } = render(<Box color='blue' />)
    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        <div
          class="cMczSr"
        >
          <div
            class="inmfTJ"
            color="blue"
          />
        </div>
      </DocumentFragment>
    `)
    // expect(json).toHaveStyleRule('color', theme.colors.blue)
  })

  it('boxShadowSize prop sets box-shadow', () => {
    const { asFragment } = render(<Box boxShadowSize='sm' />)
    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        <div
          class="cMczSr"
        >
          <div
            class="liWhWl"
          />
        </div>
      </DocumentFragment>
    `)
    // expect(json).toHaveStyleRule('box-shadow', theme.shadows.sm)
  })
})
