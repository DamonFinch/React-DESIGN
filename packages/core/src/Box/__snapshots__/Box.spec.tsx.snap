// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box > boxShadowSize prop sets box-shadow 1`] = `
<DocumentFragment>
  <div
    class="cMczSr"
  >
    <div
      class="liWhWl"
    />
  </div>
</DocumentFragment>
`;

exports[`Box > color prop sets color 1`] = `
<DocumentFragment>
  <div
    class="cMczSr"
  >
    <div
      class="inmfTJ"
      color="blue"
    />
  </div>
</DocumentFragment>
`;

exports[`Box > m prop sets margin 1`] = `
<DocumentFragment>
  <div
    class="cMczSr"
  >
    <div
      class="bAeKOk"
    />
  </div>
</DocumentFragment>
`;

exports[`Box > maxHeight, maxWidth, minHeight, minWidth 1`] = `
<DocumentFragment>
  <div
    class="cMczSr"
  >
    <div
      class="gXVrDr"
    />
  </div>
</DocumentFragment>
`;

exports[`Box > p prop sets padding 1`] = `
<DocumentFragment>
  <div
    class="cMczSr"
  >
    <div
      class="hgdKtj"
    />
  </div>
</DocumentFragment>
`;

exports[`Box > renders 1`] = `
<DocumentFragment>
  <div
    class="cMczSr"
  >
    <div
      class="cETxcT"
    />
  </div>
</DocumentFragment>
`;

exports[`Box > width and height props set width/height 1`] = `
<DocumentFragment>
  <div
    class="cMczSr"
  >
    <div
      class="gTAEUK"
      height="50%"
      width="0.5"
    />
  </div>
</DocumentFragment>
`;
